### PR TITLE
feat(css/parser): forgiving selector parsing

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Improve parsing of special pseudo classes - `:is()`/`:has()`/`:where()`

https://www.w3.org/TR/selectors-4/#forgiving-selector

**BREAKING CHANGE:**

Yes

**Related issue (if exists):**

No

### CODE REVIEW

File: crates/swc_css_parser/src/parser/selector/mod.rs [hunk 1]
```diff
@@ -140,6 +222,88 @@ where
...
+            if !eat!(self, ",") {
+                break;
+            }
+
+            let child = parse_forgiving_relative_selector(self)?;
+
+            children.push(child);
+        }
+
+        let start_pos = match children.first() {
```
Comment (kdy1) [inline crates/swc_css_parser/src/parser/selector/mod.rs:275]: I think we can use `.span_lo()` of `Spanned` here

File: crates/swc_css_parser/src/parser/selector/mod.rs [hunk 2]
```diff
@@ -140,6 +222,88 @@ where
...
+            })) => span.lo,
+            Some(ForgivingRelativeSelector::ListOfComponentValues(ListOfComponentValues {
+                span,
+                ..
+            })) => span.lo,
+            _ => {
+                unreachable!();
+            }
+        };
+        let last_pos = match children.last() {
```
Comment (kdy1) [inline crates/swc_css_parser/src/parser/selector/mod.rs:287]: Similarly we can use `span_hi()`

File: crates/swc_css_parser/src/parser/selector/mod.rs [hunk 3]
```diff
@@ -52,6 +52,88 @@ where
...
+            }
+            Some(ForgivingComplexSelector::ListOfComponentValues(ListOfComponentValues {
+                span,
+                ..
+            })) => span.lo,
+            _ => {
+                unreachable!();
+            }
+        };
+        let last_pos = match children.last() {
```
Comment (kdy1) [inline crates/swc_css_parser/src/parser/selector/mod.rs:117]: Same here

File: crates/swc_css_parser/src/parser/selector/mod.rs [hunk 4]
```diff
@@ -52,6 +52,88 @@ where
...
+            if !eat!(self, ",") {
+                break;
+            }
+
+            let child = parse_forgiving_complex_selector(self)?;
+
+            children.push(child);
+        }
+
+        let start_pos = match children.first() {
```
Comment (kdy1) [inline crates/swc_css_parser/src/parser/selector/mod.rs:105]: Same here

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you so much and sorry for a late review.

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_css_ast --breaking

**Comment by alexander-akait** [comment]
@kdy1 Fixed

**Review by kdy1 — APPROVED** [review]
Thank you!

**Review by kdy1 — APPROVED** [review]

**Pull request merged**